src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs

```diff
@@ -959,7 +978,15 @@ private bool ParseFile(string[] args, ref int i, bool noexitSeen)
                     }
                     else
                     {
-                        _collectedArgs.Add(new CommandParameter(null, arg));
+                        bool? boolValue = GetBoolValue(arg);
```

Comment:
You can still simplify though, something like:
```c#
object argValue = arg.Substring(offset + 1);
 _collectedArgs.Add(new CommandParameter(arg.Substring(0, offset), GetArgPossiblyAsBool(argValue)));
```

And `GetArgPossiblyAsBool` returns `object`.